lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -77,31 +78,26 @@ sub _save_vars {
     close($fd);
 }
 
-# When changing something here, also take a look at OpenQA::Utils::asset_type_from_setting
 sub detect_asset_keys {
     my ($vars) = @_;
 
     my %res;
-    for my $isokey (qw(ISO), map { "ISO_$_" } (1 .. 9)) {
-        $res{$isokey} = 'iso' if $vars->{$isokey};
+    my $type;
+
+    for my $key (keys(%$vars)) {
+        # UEFI_PFLASH_VARS may point to an image uploaded by a previous
+        # test (which we should treat as an hdd asset), or it may point
+        # to an absolute filesystem location of e.g. a template file from
+        # edk2 (which we shouldn't).
+        next if $key eq 'UEFI_PFLASH_VARS' && $vars->{$key} =~ m,^/,;
```

Comment:
I'd prefer to store `$vars->{$key}` in a variable.

Reply:
Fixed.